lib/trino-parquet/src/main/java/io/trino/parquet/ParquetReaderOptions.java

```diff
@@ -23,11 +24,13 @@
     private static final DataSize DEFAULT_MAX_READ_BLOCK_SIZE = DataSize.of(16, MEGABYTE);
     private static final DataSize DEFAULT_MAX_MERGE_DISTANCE = DataSize.of(1, MEGABYTE);
     private static final DataSize DEFAULT_MAX_BUFFER_SIZE = DataSize.of(8, MEGABYTE);
+    private static final DataSize DEFAULT_MAX_VECTOR_LENGTH = DataSize.of(8, KILOBYTE);
 
     private final boolean ignoreStatistics;
     private final DataSize maxReadBlockSize;
     private final DataSize maxMergeDistance;
     private final DataSize maxBufferSize;
+    private final DataSize maxVectorLength;
```

Comment:
Please add equivalent session properties to hive, delta and iceberg since we have the related `parquet_max_read_block_size` also available. Having session property is also necessary for users to assess impact easily.